backend/internal/authz/engine.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package authz
+
+// authorizationEngine is the internal interface for authorization engines.
+// This interface is NOT exported and is used internally by the authorization service.
+// Different engines can be plugged in (RBAC, ABAC, ReBAC, Custom) by implementing this interface.
+type authorizationEngine interface {
```

Comment:
I'd suggest to make this public and let rbac engine implement this. Currently we cannot guarantee engine impl implements the authorizationEngine interface right?